proposals/records-wip.md

```diff
@@ -28,3 +28,35 @@ struct_body
 ```
 
 The `attributes` non-terminal will also permit a new contextual attribute, `data`.
+
+A class (struct) declared with a parameter list or `data` modifier is called a record class (record struct), either of which is a record type.
+
+It is an error to declare a record type without both a parameter list and the `data` modifier.
```

Comment:
Is there any consideration of splitting these two? So a parameter lost on its own is a primary constructor, and data causes it to generate properties and equality.

Reply:
The reason the `data` keyword is being introduced is to keep the door open to doing that.  We'll be discussing next week whether or not we want to stick with that (or, alternatively, drop the `data` keyword).

---
In reply to: [364061435](https://github.com/dotnet/csharplang/pull/3076#discussion_r364061435) [](ancestors = 364061435)